### PR TITLE
refactor(server): removes teardown option

### PR DESCRIPTION
**Pull request by ixahmedxi**
Closes #

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

As discussed previously, this PR removes the teardown function option.

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

**Review by jlalmes — APPROVED** [review]
LGTM - have made a note to remove `teardown` in `next` branch here too: https://github.com/jlalmes/trpc-openapi/issues/156

**Review by KATT — APPROVED** [review]
It just got some merge conflict now!

File: examples/.interop/next-prisma-todomvc/src/pages/api/trpc/[trpc].ts [hunk 1]
```diff
@@ -33,11 +33,10 @@ export type AppRouter = typeof router;
...
   createContext,
-  teardown: () => prisma.$disconnect(),
   onError({ error }) {
     if (error.code === 'INTERNAL_SERVER_ERROR') {
       // send to bug reporting
       console.error('Something went wrong', error);
     }
   },
-});
+});
```
Comment (KATT) [inline examples/.interop/next-prisma-todomvc/src/pages/api/trpc/[trpc].ts:42]: WAT

Comment (KATT) [inline examples/.interop/next-prisma-todomvc/src/pages/api/trpc/[trpc].ts:42]:
```suggestion
});
```

Comment (ixahmedxi) [reply examples/.interop/next-prisma-todomvc/src/pages/api/trpc/[trpc].ts:42]: oh wtf

Comment (ixahmedxi) [reply examples/.interop/next-prisma-todomvc/src/pages/api/trpc/[trpc].ts]: goddamit

**Review by KATT — APPROVED** [review]
Could you add a note on the migration page somewhere? Just so we don't forget to mention it

**Pull request merged**